### ISSUE TITLE
postgresql - to_tsquery docs and implementation detail

### ISSUE THREAD

**Issue by sqlalchemy-bot** [sqlalchemy/sqlalchemy#3160]
**Migrated issue, originally created by jvanasco ([@jvanasco](https://github.com/jvanasco))**

I recently realized some implementation details of `to_tsquery` that are incompatible with the docs I drafted and how sqlalchemy integrates it.

Not sure how to handle this.

I based the docs/examples on existing tests and text.  So we have this as the first bit of "Full Text Search"  ( http://docs.sqlalchemy.org/en/latest/dialects/postgresql.html#full-text-search )

    select([sometable.c.text.match("search string")])
    SELECT text @@ to_tsquery('search string') FROM table

well, if we put this into psql...

    badsql=> select to_tsquery('search string') ;
    ERROR:  syntax error in tsquery: "search string"

that's because tsquery has a rigid enforcement of input.  text must either be tokenized and joined with acceptable operators :

    select to_tsquery('cat & rat');

be quoted :

    select to_tsquery('''search string''');

or use the alternate function

    select plainto_tsquery('search string');

So, these are acceptable:

    select to_tsquery('search & string');
    select to_tsquery('''search string''');
    select plainto_tsquery('search string');

but this is not:

    select to_tsquery('search string');

I'm not sure the best way to handle this nuance in the docs.

As far as the implementation detail goes... 

this creates issues with `Column.match`, because that generates (invalid) sql like this:

    table.column @@ to_tsquery('search string')

not valid sql  like:

    table.column @@ to_tsquery('''search string''')
    table.column @@ plainto_tsquery('search string')

This is an annoying implementation detail , but the content/type of string will affect the function or format t search on.

i thought about just regexing this into submission, but the multitude of possible operators and edge cases suggests that the text would need to be parsed for tokenization instead -- otherwise throwing in a bit of text that has an operator will function as a real operator and probably trigger an invalid syntax.  

i think the easiest scenario would be to replace match's `to_tsquery` with `plainto_tsquery`

**Comment by MajorDallas** [comment]
I've just run into this, but in my case I'm wanting to use the `websearch_to_tsquery` function.

Given the many ways for a postgres user to transform a bit of text into a `tsquery`, what I would want as a user of SQLAlchemy is the option to give `.match()` the `tsquery` I want to use myself rather than have it lock me in to just one pathway, whether that's `to_tsquery` or `plain_to_tsquery`.

Ex:

```python
my_table.c.tsvec_col.match(func.websearch_to_tsquery('"foo bar"'))
# or
my_table.c.tsvec_col.match('"foo bar"', postgres_tsquery_factory=func.websearch_to_tsquery)
```
emitting
```sql
SELECT * FROM my_table WHERE tsvec_col @@ websearch_to_tsquery('"foo bar"');
```

I think this shouldn't get in the way of other backends since

* in the first example, it's simply passing exactly what the user gives, and
* in the second example, we're using a kwarg that can be as dialect-specific or -agnostic as appropriate (went with specific for my example).

Then, `.match()` should emit `plain_to_tsquery()` as a fallback if it receives a single `str`.

**Comment by jvanasco** [comment]
@MajorDallas I descoped your request into a new ticket for the new functionality. This ticket is deep in a backlog of old documentation issues, and I'd like to close it out as a purely documentation update.

**Comment by sqla-tester** [comment]
**jonathan vanasco** has proposed a fix for this issue in the **master** branch:

**Fixes: #3160** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3099
